deploy/injector-rbac.yaml

```diff
@@ -38,3 +38,36 @@ subjects:
 - kind: ServiceAccount
   name: vault-injector
   namespace: vault
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: vault-injector-role
+  labels:
+    app.kubernetes.io/name: vault-injector
+    app.kubernetes.io/instance: vault
+rules:
+- apiGroups: [""]
+  resources: ["endpoints", "secrets"]
+  verbs: 
+    - "create"
```

Comment:
Is `create` needed if `deploy/injector-leader-extras.yaml` is deployed? Is this to prevent a race or to allow it to still work if users don't deploy that file?

Reply:
I think it is, since the leader-elector needs to be able to recreate the Endpoint if it is deleted, and similarly the vault-k8s leader will create the Secret if it doesn't exist when it generates a certificate.

Reply:
@tvoran is correct, this is needed.